src/Chunk.cpp

```diff
@@ -1864,13 +1863,41 @@ void cChunk::RemoveClient(cClientHandle * a_Client)
 		}
 	}
 
+	// All clients were removed. No one sees this chunk any more.
+	// Delete all hostiles inside it.
+	if (!HasAnyClients())
+	{
+		for (cEntityList::iterator itr = m_Entities.begin(); itr != m_Entities.end(); ++itr)
+		{
+			cEntity * Entity = *itr;
+			if (Entity->IsTicking() && Entity->IsMob())
+			{
+				auto Monster = static_cast<cMonster *>(Entity);
+				if (
+					(Monster->GetMobFamily() == cMonster::eFamily::mfHostile) &&
+					(Monster->GetCustomName() == "")  // Monsters that have name tags are never removed
```

Comment:
The writers of `std::string::empty` express their dissatisfaction at the non-use of this function here.